transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala

```diff
@@ -99,24 +113,78 @@ object HcaPipelineBuilder extends PipelineBuilder[Args] {
     * @return a Msg object in the desired output format
     */
   def transformMetadata(entityType: String, fileName: String, metadata: Msg): Msg = {
-    // extract info from file name
+    val (entityId, entityVersion) = getEntityIdAndVersion(fileName: String)
+    Obj(
+      mutable.LinkedHashMap[Msg, Msg](
+        Str(s"${entityType}_id") -> Str(entityId),
+        Str("version") -> Str(entityVersion),
+        Str("content") -> Str(encode(metadata).getOrElse(""))
```

Comment:
Oof, my mistake for missing this in the original code. `getOrElse("")` is bad (it hides a lack of data). We should either:
1. Throw an exception here if `content` is truly required, or
2. Have `content` be nullable, and leave it out when `encode(metadata)` returns `None`

Under what circumstances would we expect `encode` to return `None`?

Reply:
Then `encode` method only returns none when the metadata Msg is empty. Would it make sense to update the `encode` method itself to throw an exception when its input is empty, and then change its output type to be a string (instead of an `Option[String]`)?

Reply:
In this case, I don't think we need to check emptiness of the input object at all. If it's an empty object then the desired output is `"{}"`

Reply:
Sounds good. I just updated that method and then wrote another quick test to make sure it was actually outputing the `{}`.